### DISCUSSION TITLE
How to add custom weight for audit ref  ids for categories

### DISCUSSION THREAD

**Discussion by ashishkshrivastava** [GoogleChrome/lighthouse#11735] [Q&A] [answered]
Hi

I am trying to add custom weight for **audit ref**  ids in categories (performance) using a custom config js. I would like to calculate performance on custom weight instead of using the default lighthouse weight. I also observe after adding the below change , in the report all the **audit ref** duplicated. I am using lighthouse version 6.

Below are the details of my custom config.

```js
module.exports = {
  extends: 'lighthouse:default',
  categories: {
    performance: {
      title: 'Performance',
      description: 'This category judges your performance',
      auditRefs: [
        {id: 'total-blocking-time', weight: 10, group: 'metrics'},
        {id: 'first-contentful-paint', weight: 20, group: 'metrics'},
        {id: 'interactive', weight: 25, group: 'metrics'},
        {id: 'speed-index', weight: 25, group: 'metrics'},
        {id: 'largest-contentful-paint', weight: 5, group: 'metrics'},
        {id: 'cumulative-layout-shift', weight: 15, group: 'metrics'}
      ],
    }
  }
};
```

Thanks & Regards
Ashish

**Comment by patrickhulce** [comment, marked as answer]
If you want your category to *replace* the default performance category, then you can't extend the default config. Extension for categories is *additive*.

tl;dr remove `extends: 'lighthouse:default'` if you want to replace a category.

**Comment by ashishkshrivastava** [comment]
**After removing the extends: 'lighthouse:default' I am getting error in pipeline.**

```
01 Dec 2020 16:21:00 GMT ChromeLauncher Killing Chrome instance 82
Runtime error encountered: could not find total-blocking-time audit for category performance
Error: **could not find total-blocking-time audit for category performance**
    at /usr/local/lib/node_modules/@lhci/cli/node_modules/lighthouse/lighthouse-core/config/config.js:125:15
    at Array.forEach (<anonymous>)
    at /usr/local/lib/node_modules/@lhci/cli/node_modules/lighthouse/lighthouse-core/config/config.js:118:38
    at Array.forEach (<anonymous>)
    at assertValidCategories (/usr/local/lib/node_modules/@lhci/cli/node_modules/lighthouse/lighthouse-core/config/config.js:117:27)
```

Please guide me if i am missing any other configuration

**Comment by patrickhulce** [comment]
Well yes that's not the *only* change to make, if you're not extending anymore but still want to use those audits you'll need to [copy what you need from the default config](https://github.com/GoogleChrome/lighthouse/blob/master/lighthouse-core/config/default-config.js). ~~Those sorts of questions and troubleshooting are best left for another forum though :)~~ (moved from issue to discussions)

**Comment by ashishkshrivastava** [comment]
Thanks for your response.
